lib/networks/networks_list.go

```diff
@@ -0,0 +1,1101 @@
+	ChainID_TELOS                   uint64 = 40
+	ChainID_TELOS_TESTNET           uint64 = 41
+	ChainID_KOVAN                   uint64 = 42
+	ChainID_BSC                     uint64 = 56
+	ChainID_OPTIMISM_KOVAN          uint64 = 69
+	ChainID_BSC_TESTNET             uint64 = 97
+	ChainID_GNOSIS                  uint64 = 100
+	ChainID_POLYGON                 uint64 = 137
+	ChainID_MONAD                   uint64 = 143
+	ChainID_OPTIMISM_GOERLI         uint64 = 420
+	ChainID_POLYGON_ZKEVM           uint64 = 1101
+	ChainID_MOONBEAM                uint64 = 1284
+	ChainID_MOONBASE_ALPHA          uint64 = 1287
+	ChainID_SEI_TESTNET             uint64 = 1328
+	ChainID_SEI                     uint64 = 1329
+	ChainID_SONEIUM                 uint64 = 1868
+	ChainID_SONEIUM_MINATO          uint64 = 1946
+	ChainID_B3_SEPOLIA              uint64 = 1993
+	ChainID_SOMNIA                  uint64 = 5031
+	ChainID_SANDBOX_TESTNET         uint64 = 6252
+	ChainID_B3                      uint64 = 8333
+	ChainID_BASE                    uint64 = 8453
+	ChainID_MONAD_TESTNET           uint64 = 10143
+	ChainID_INCENTIV_TESTNET        uint64 = 11690
+	ChainID_IMMUTABLE_ZKEVM         uint64 = 13371
+	ChainID_IMMUTABLE_ZKEVM_TESTNET uint64 = 13473
+	ChainID_HOMEVERSE               uint64 = 19011
+	ChainID_INCENTIV_TESTNET_V2     uint64 = 28802
+	ChainID_HARDHAT                 uint64 = 31337
+	ChainID_HARDHAT2                uint64 = 31338
+	ChainID_APECHAIN_TESTNET        uint64 = 33111
+	ChainID_APECHAIN                uint64 = 33139
+	ChainID_HOMEVERSE_TESTNET       uint64 = 40875
+	ChainID_ARBITRUM                uint64 = 42161
+	ChainID_ARBITRUM_NOVA           uint64 = 42170
+	ChainID_ETHERLINK               uint64 = 42793
+	ChainID_AVALANCHE_TESTNET       uint64 = 43113
+	ChainID_AVALANCHE               uint64 = 43114
+	ChainID_SOMNIA_TESTNET          uint64 = 50312
+	ChainID_MUMBAI                  uint64 = 80001
+	ChainID_AMOY                    uint64 = 80002
+	ChainID_BLAST                   uint64 = 81457
+	ChainID_BASE_GOERLI             uint64 = 84531
+	ChainID_BASE_SEPOLIA            uint64 = 84532
+	ChainID_BORNE_TESTNET           uint64 = 94984
+	ChainID_ETHERLINK_TESTNET       uint64 = 128123
+	ChainID_ARBITRUM_GOERLI         uint64 = 421613
+	ChainID_ARBITRUM_SEPOLIA        uint64 = 421614
+	ChainID_XAI                     uint64 = 660279
+	ChainID_KATANA                  uint64 = 747474
+	ChainID_ARC_TESTNET             uint64 = 5042002
+	ChainID_SEPOLIA                 uint64 = 11155111
+	ChainID_OPTIMISM_SEPOLIA        uint64 = 11155420
+	ChainID_TOY_TESTNET             uint64 = 21000000
+	ChainID_SKALE_NEBULA_TESTNET    uint64 = 37084624
+	ChainID_BLAST_SEPOLIA           uint64 = 168587773
+	ChainID_SKALE_NEBULA            uint64 = 1482601649
+	ChainID_XAI_SEPOLIA             uint64 = 37714555429
+)
+
+// All supported networks
+var all = Networks{
+	"mainnet": &Network{
+		ChainID: ChainID_MAINNET,
+		Name:    "Ethereum",
```

Comment:
I'm thinking you'd reference an existing network here -- as this is just a mapping, right?
```go
var all = Networks{
	"mainnet": MAINNET,
	"rposten": ROPSTEN,
```

```go
var (
	MAINNET = &Network{}
	ROPSTEN = &Network{}
)
```